numpy/_core/tests/test_strings.py

```diff
@@ -126,6 +126,31 @@ def test_add(self, in1, in2, out, dt):
         out = np.array(out, dtype=dt)
         assert_array_equal(np.strings.add(in1, in2), out)
 
+    @pytest.mark.parametrize("in1,in2,out", [
+        ("abc", 3, "abcabcabc"),
+        ("abc", 0, ""),
+        ("abc", -1, ""),
+        (["abc", "def"], [1, 4], ["abc", "defdefdefdef"]),
+    ])
+    def test_multiply(self, in1, in2, out, dt):
+        in1 = np.array(in1, dtype=dt)
+        out = np.array(out, dtype=dt)
+        assert_array_equal(np.strings.multiply(in1, in2), out)
+
+    def test_multiply_raises(self, dt):
+        with pytest.raises(TypeError, match="unsupported type"):
+            np.strings.multiply(np.array("abc", dtype=dt), 3.14)
+
+        with pytest.raises(MemoryError):
+            np.strings.multiply(np.array("abc", dtype=dt), sys.maxsize)
+
+    @pytest.mark.parametrize("i_dt", [np.int8, np.int16, np.int32, np.int64])
```

Comment:
Nice, I'm glad you got this working - can you also check python integers work too? There's a DType specifically for python integer operands and I'm not sure if that matters for the legacy ufunc setups.